src/USER-MISC/fix_charge_regulation.cpp

```diff
@@ -0,0 +1,1335 @@
+/* ----------------------------------------------------------------------
+   LAMMPS - Large-scale Atomic/Molecular Massively Parallel Simulator
+   http://lammps.sandia.gov, Sandia National Laboratories
+   Steve Plimpton, sjplimp@sandia.gov
+
+   Copyright (2003) Sandia Corporation.  Under the terms of Contract
+   DE-AC04-94AL85000 with Sandia Corporation, the U.S. Government retains
+   certain rights in this software.  This software is distributed under
+   the GNU General Public License.
+
+   See the README file in the top-level LAMMPS directory.
+------------------------------------------------------------------------- */
+
+/* ----------------------------------------------------------------------
+   Contributing author: Tine Curk (tcurk5@gmail.com) and Jiaxing Yuan (yuanjiaxing123@hotmail.com)
+------------------------------------------------------------------------- */
+#include "fix_charge_regulation.h"
+#include <mpi.h>
+#include <cmath>
+#include <cstring>
+#include "atom.h"
+#include "atom_vec.h"
+#include "molecule.h"
+#include "update.h"
+#include "modify.h"
+#include "fix.h"
+#include "comm.h"
+#include "compute.h"
+#include "group.h"
+#include "domain.h"
+#include "region.h"
+#include "random_park.h"
+#include "force.h"
+#include "pair.h"
+#include "bond.h"
+#include "angle.h"
+#include "dihedral.h"
+#include "improper.h"
+#include "kspace.h"
+#include "math_extra.h"
+#include "math_const.h"
+#include "memory.h"
+#include "error.h"
+#include "neighbor.h"
+
+using namespace LAMMPS_NS;
+using namespace FixConst;
+using namespace MathConst;
+
+// large energy value used to signal overlap
+#define MAXENERGYSIGNAL 1.0e100
+#define MAXENERGYTEST 1.0e50
+#define small 0.0000001
+#define PI 3.1415926
```

Comment:
Please do not define your own (and rather imprecise) version of PI and use MY_PI from the MathConst namespace instead.